warp-tls/Network/Wai/Handler/WarpTLS.hs

```diff
@@ -323,12 +321,8 @@ mkConn
     -> params
     -> IO (Connection, Transport)
 mkConn tlsset set s params = do
-    var <- newEmptyMVar
-    _ <- forkIOWithUnmask $ \umask -> do
-        let tm = settingsTimeout set * 1000000
-        mct <- umask (timeout tm recvFirstBS)
-        putMVar var mct
-    mbs <- takeMVar var
+    let tm = settingsTimeout set * 1000000
+    mbs <- timeout tm recvFirstBS
```

Comment:
I think this makes sense, but if we want to keep the same behaviour we'd have to change it to:
```hs
mbs <- unsafeUnmask $ timeout tm recvFirstBS
```

Reply:
The original code before unsafeUnmask does not have `unsafeUnmask`.
Is it really necessary?

Reply:
The `umask` in the diff is basically `unsafeUnmask`. But as you described earlier, that `mkConn` should be interruptable, this is unnecessary if the `bracket` (or whatever is used) is from `Control.Exception` instead of from `UnliftIO`.

Because the whole `forkWithUnmask` was a hack to get around the `unliftio` functions that `mask` more than necessary.

So no, I have changed my stance and adding `unsafeUnmask` is indeed not necessary.